Plot action to switch on/off OpenGL

vallsv opened the pull request:
Following #3255, this PR provides a plot action which can be used with a silx plot.

It have to be setup manually.

It only shows if OpenGL is enabled or not. I guess it is already complex for most people.

That's more or less the way i will use it in Flint.

It can be used with this example.

```
./bootstrap.py examples/plotInteractiveImageROI.py
```

![Screenshot from 2020-11-10 15-00-38](https://user-images.githubusercontent.com/7579321/98684937-11abaf80-2367-11eb-9919-21e64888ada9.png)

Changelog: 
- Added an action to switch on/off OpenGL rendering on a plot

t20100 commented on silx/gui/plot/actions/control.py:
```diff
@@ -632,3 +632,73 @@ def _modeChanged(self, source):
         self.plot._eventHandler.validate()
+
+
+class OpenGlAction(PlotAction):
```
Best renamed to `OpenGLAction` to follow naming convention.

vallsv replied:
This one is fixed

t20100 commented on silx/gui/plot/actions/control.py:
```diff
@@ -632,3 +632,73 @@ def _modeChanged(self, source):
+        super(OpenGlAction, self).__init__(
+            plot,
+            icon=icon,
+            text='Toggle keep aspect ratio',
```
`text` was not updated after copy-paste I guess.

t20100 commented on silx/gui/plot/actions/control.py:
```diff
@@ -632,3 +632,73 @@ def _modeChanged(self, source):
+            return
+        if name is not "opengl":
+            from silx.gui.utils import glutils
+            result = glutils.isOpenGLAvailable()
```
I would make the test at action creation, but I guess you want to keep it fast.
Once it failed once, you can disable the action, it will never become available.

vallsv replied:
Yes i don't want crazy tests at creation.

I don't know about disabling anyway. I prefer not to do so.

If you click on it again you have an obvious error message, you don't need to make it safer.
If you are able to install `pyopengl` in your env, the problem could be fixed without restarting the application.

t20100 replied:
It would work for install `pyopengl`, but for any other error, it would not since `isOpenGLAvailable` caches the runtime check result.

vallsv replied:
That's right, we could add a parameter to skip the cache here.

t20100 replied:
Well, there is no reason the result of the runtime check would change...
`pyopengl` availability, `$DISPLAY` and Qt reporting an issue are checked each time, only the runtime check which is run after those first checks is cached.

vallsv replied:
Driver update, or lib update

t20100 replied:
without restarting the app, things will already be loaded, no?

t20100 reviewed (approved):
I left a few minor comments, else LGTM

vallsv commented:
I have reworked the PR in order fix the text and the name.

I have also removed the disabled icon, and make it checkable. This allow to use it as it is with a menu.
